app/Domain/Integrations/Models/IntegrationModel.php

```diff
@@ -93,6 +93,7 @@ public function activateWithCoupon(string $couponCode): void
     {
         $coupon = CouponModel::query()
             ->where('code', '=', $couponCode)
+            ->where('is_distributed', '=', true)
```

Comment:
I think when doing this as an admin setting it to `distributed` is not needed. Although the field could be set to true when doing the activation.